src/mongo/db/exec/benchmark_test.cpp

```diff
@@ -0,0 +1,254 @@
+ *
+ *    This program is free software: you can redistribute it and/or  modify
+ *    it under the terms of the GNU Affero General Public License, version 3,
+ *    as published by the Free Software Foundation.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *    As a special exception, the copyright holders give permission to link the
+ *    code of portions of this program with the OpenSSL library under certain
+ *    conditions as described in each individual source file and distribute
+ *    linked combinations including the program with the OpenSSL library. You
+ *    must comply with the GNU Affero General Public License in all respects for
+ *    all of the code used other than as permitted herein. If you modify file(s)
+ *    with this exception, you may extend this exception to your version of the
+ *    file(s), but you are not obligated to do so. If you do not wish to do so,
+ *    delete this exception statement from your version. If you delete this
+ *    exception statement from all source files in the program, then also delete
+ *    it in the license file.
+ */
+
+/**
+ * This file contains a sample usage of TEST_F for benchmarking
+ */
+
+
+#include "mongo/db/exec/working_set.h"
+#include "mongo/db/jsobj.h"
+#include "mongo/db/json.h"
+#include "mongo/db/storage/snapshot.h"
+#include "mongo/platform/random.h"
+#include "mongo/unittest/unittest.h"
+#include "mongo/util/assert_util.h"
+#include "mongo/util/string_map.h"
+#include "mongo/util/time_support.h"
+
+#include <unordered_map>
+
+#define TRIALS 100000
+#define STR_LEN 32
+#define OUTFILE "benchmarks.json"
+
+namespace mongo {
+
+using std::string;
+
+std::string gen_random(const int);
+
+PseudoRandom* rand = new PseudoRandom((uint32_t)time(NULL));
+
+const auto stdStringList = ([]() -> std::vector<std::string> {
+    std::vector<std::string> strings;
+    for (int i = 0; i < TRIALS * 2; i++) {
+        strings.push_back(gen_random(STR_LEN));
+    }
+    return strings;
+})();
+
+const auto cStringList = ([]() -> std::vector<const char*> {
+    std::vector<const char*> strings;
+    for (int i = 0; i < TRIALS * 2; i++) {
+        strings.push_back(stdStringList[i].c_str());
+    }
+    return strings;
+})();
+
+// http://stackoverflow.com/questions/440133/how-do-i-create-a-random-alpha-numeric-string-in-c
+std::string gen_random(int length) {
+    static string charset = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890";
+    string result;
+    result.resize(length);
+    for (int i = 0; i < length; i++) {
+        result[i] = charset[rand->nextInt32(charset.length())];
```

Comment:
It turns out that the operator[] in std::string can have a surprising amount of overhead. It would be better to use a range-for loop like for (char& c : result) { c = something(); }